std/typecons.d

```diff
@@ -7407,7 +7313,89 @@ private template rtFieldDispatch(bool set, Field, bool refField, Obj, bool refOb
             }
         }
 
-        assert (0, "`~ errorMessage ~`");
+        assert (0, "` ~ Obj.stringof ~ ` has no ` ~ (set ? `assignable` : (refField ? `lvalue ` : ``)) ~ `
+            property or field with the specified name matching ` ~ Field.stringof ~ `");
     } `;
+
+public:
     mixin(dispMix);
 }
+
+@safe nothrow pure unittest
+{
+    enum A =
+`   @safe: @nogc: nothrow: pure:
+    private:
+        int _valProp = 31;
+        int _refProp = -78;
+
+    public:
+        int field = 564;
+        
+    @property:
+        int valProp() const
+        {
+            return _valProp;
+        }
+        void valProp(int val)
+        {
+            _valProp = val;
+        }
+        
+        ref inout(int) refProp() inout
+        {
+            return _refProp;
+        }`;
+
+    static struct S
+    {
+        mixin(A);
+    }
+
+    static class C
+    {
+        mixin(A);
+    }
+
+    static void testRead(Obj)(const Obj obj) @nogc
+    {
+        assert(obj.getField!int("valProp") == 31);
+        assert(obj.getField!int("refProp") == -78);
+        assert(obj.getField!int("field") == 564);
+        
+        int got1, got2, got3;
+        obj.getField("valProp", got1);
+        assert(got1 == 31);
+        obj.getField("refProp", got2);
+        assert(got2 == -78);
+        obj.getField("field", got3);
+        assert(got3 == 564);
+        
+        //assertThrown(obj.refField!(const int)("valProp"));
```

Comment:
@rcorre Since the name lookup is done at run-time, rather than compile-time, a failed call triggers the `assert(0)` on line 7316. Unfortunately, there is no reliable way to catch an `assert(0)` error.

Originally I designed `rtFieldDispatch()` to just throw a regular `Exception` which could be caught, but **JackStouffer** wants this stuff to be `nothrow @nogc` if possible.